ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/ParseReportingUrl.java

```diff
@@ -176,34 +224,30 @@ public Result<PCollection<PubsubMessage>, PubsubMessage> expand(
           }
 
           reportingUrl = urlParser.toString();
-
-          attributes.put(Attribute.REPORTING_URL, reportingUrl);
-          payload.put(Attribute.REPORTING_URL, reportingUrl);
-
           PerDocTypeCounter.inc(attributes, "valid_url");
-          return new PubsubMessage(Json.asBytes(payload), attributes);
+          return interaction.toBuilder().setReportingUrl(reportingUrl).build();
         }).exceptionsInto(TypeDescriptor.of(PubsubMessage.class))
-            .exceptionsVia((ExceptionElement<PubsubMessage> ee) -> {
-              try {
-                throw ee.exception();
-              } catch (UncheckedIOException | IllegalArgumentException
-                  | ParsedReportingUrl.InvalidUrlException | RejectedMessageException e) {
-                return FailureMessage.of(ParseReportingUrl.class.getSimpleName(), ee.element(),
-                    ee.exception());
-              }
-            }));
+        .exceptionsVia((ExceptionElement<PubsubMessage> ee) -> {
+          try {
+            throw ee.exception();
+          } catch (UncheckedIOException | IllegalArgumentException
+              | ParsedReportingUrl.InvalidUrlException | RejectedMessageException e) {
+            return FailureMessage.of(ParseReportingUrl.class.getSimpleName(), ee.element(),
+                ee.exception());
+          }
+        }));
   }
 
   @VisibleForTesting
-  boolean isUrlValid(URL url, String doctype) {
+  boolean isUrlValid(URL url, String interaction) {
```

Comment:
this was a bit confusing to see `interaction` be a `String`, and based on how this is called I would expect this parameter to be `interactionType`